lib/Service/ContextService.php

```diff
@@ -252,6 +252,19 @@ public function delete(int $contextId, string $userId): Context {
 		return $context;
 	}
 
+	public function deleteNodeRel(int $nodeId, int $nodeType): void {
+		try {
+			$nodeRelIds = $this->contextNodeRelMapper->getRelIdsForNode($nodeId, $nodeType);
+			foreach ($nodeRelIds as $nodeRelId) {
+				$this->pageContentMapper->deleteByNodeRelId($nodeRelId);
+				$this->contextNodeRelMapper->deleteByNodeRelId($nodeRelId);
+			}
```

Comment:
We can save the n+1 queries here by rather having methods that support multiple ids in an array and use an `IN()` sql statement. Might not be crucial here but generally a good pattern to follow for performance instead of doing queries in a loop

Reply:
Not sure how to go about this :confused: :confused:

Reply:
Let's check that together next week then :)